src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +39,655 @@ const ActorInit En_Rz_InitVars = {
+
+            if (CHECK_FLAG_ALL(this->actor.flags, ACTOR_FLAG_10000)) {
+                this->actionFunc = func_80BFC36C;
+                this->actor.cutscene = this->cutscenes[0];
+                this->actor.flags &= ~ACTOR_FLAG_10000;
+            } else if (Player_GetMask(play) == PLAYER_MASK_KAMARO) {
+                if (gSaveContext.save.weekEventReg[77] & 4) {
+                    Message_StartTextbox(play, 0x2925, &this->actor);
+                    func_80BFBDFC(play);
+                } else {
+                    Message_StartTextbox(play, 0x2926, &this->actor);
+                    func_80BFBDFC(play);
+                }
+            } else {
+                Message_StartTextbox(play, 0x2926, &this->actor);
+                func_80BFBDFC(play);
+            }
+
+        } else if (EnRz_CanTalk(this, play)) {
+            if (func_80BFBCEC(this, play) && !(gSaveContext.save.weekEventReg[77] & 4) && this->unk_40C != NULL) {
+                this->actor.flags |= ACTOR_FLAG_10000;
+                func_800B8500(&this->actor, play, 1000.0f, 1000.0f, -1);
+            } else {
+                this->actor.flags &= ~ACTOR_FLAG_10000;
+                func_800B8614(&this->actor, play, 120.0f);
+            }
+        }
+
+        if (!EN_RZ_GET_PARAM_8000(&this->actor)) {
+            sp30.x = this->actor.projectedPos.x;
+            sp30.y = this->actor.projectedPos.y;
+            sp30.z = this->actor.projectedPos.z;
+            func_801A1FB4(3, &sp30, 0x2A, 900.0f);
+        }
+    }
+}
+
+void func_80BFC608(EnRz* this, PlayState* play) {
+    EnRz_UpdateAnimation(this, play);
+
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        func_801477B4(play);
+        this->actionFunc = func_80BFC674;
+    }
+}
+
+void func_80BFC674(EnRz* this, PlayState* play) {
+    EnRz_UpdateAnimation(this, play);
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actionFunc = func_80BFC608;
+        if (Player_GetMask(play) == PLAYER_MASK_KAMARO) {
+            Message_StartTextbox(play, 0x2925, &this->actor);
+            return;
+        }
+        Message_StartTextbox(play, 0x2924, &this->actor);
+        return;
+    }
+
+    if (EnRz_CanTalk(this, play)) {
+        func_800B8614(&this->actor, play, 120.0f);
+    }
+}
+
+void func_80BFC728(EnRz* this, PlayState* play) {
+    EnRz_UpdateAnimation(this, play);
+
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        func_801477B4(play);
+        this->actionFunc = func_80BFC7E0;
+        this->actor.textId++;
+        if (!EN_RZ_GET_PARAM_8000(&this->actor)) {
+            if (this->actor.textId >= 0x2920) {
+                this->actor.textId = 0x291C;
+            }
+        } else if (this->actor.textId >= 0x2924) {
+            this->actor.textId = 0x2920;
+        }
+    }
+}
+
+void func_80BFC7E0(EnRz* this, PlayState* play) {
+    s32 new_var;
```

Comment:
Is the temp really needed?

Reply:
i'm afraid so
![image](https://user-images.githubusercontent.com/96613413/185785521-4b786de0-da32-49b6-b16e-a44eeaf054fc.png)

Reply:
What if `EnRz_UpdateSkelAnime` returns `s32` instead of `void`?

Reply:
that fixed it, thanks